text/2603-rust-symbol-name-mangling-v0.md

```diff
@@ -746,6 +747,10 @@ Mangled names conform to the following grammar:
 // We use <path> here, so that we don't have to add a special rule for
 // compression. In practice, only a crate root is expected.
 <instantiating-crate> = <path>
+
+// There are no restrictions on the characters that may be used
+// in the suffix following the `.`.
+<vendor-specific-suffix> = "." <suffix>
```

Comment:
Can the suffix contain any characters at all? Or do these characters need to be in `A-Z`, `a-z`, `0-9`, and `_` as in [RFC 2603](https://rust-lang.github.io/rfcs/2603-rust-symbol-name-mangling-v0.html)?

Reply:
```suggestion
// The suffix following `.` can contain any character in the allowed alphanumeric set.
<vendor-specific-suffix> = "." <suffix>
```

Reply:
Since we have no control over what external parties like LLVM stick into that suffix, we should indeed strive for allowing anything at all. LLVM for example will add suffixes that contain more `.` characters, like `.llvm.3285396211802591752` or `.cold.1`.